crates/ty_python_semantic/src/types/typed_dict.rs

```diff
@@ -143,30 +143,57 @@ impl TypedDictAssignmentKind {
 pub(super) fn validate_typed_dict_key_assignment<'db, 'ast>(
```

Comment:
Calls to this function are becoming quite difficult to read because of how many arguments there are. Especially the third argument, where `None` is often passed, and the last argument which is a `bool`. We could consider refactoring this into a struct that holds all the options for better readability, e.g.

```rs
let validator = TypedDictValidator {
    context: &self.context,
    typed_dict,
    full_object_ty: None,
    key,
    value_ty,
    typed_dict_node,
    key_node,
    value_node,
    assignment_kind,
    emit_diagnostics: true
};

validator.validate();
```